porcelain/miner.go

```diff
@@ -509,3 +509,57 @@ func MinerGetCollateral(ctx context.Context, plumbing mgaAPI, minerAddr address.
 	}
 	return types.NewAttoFILFromBytes(rets[0]), nil
 }
+
+// mwapi is the subset of the plumbing.API that Miner(Set|Get)WorkerAddress use.
+type mwapi interface {
+	ConfigGet(dottedPath string) (interface{}, error)
+	MessageSend(ctx context.Context, from, to address.Address, value types.AttoFIL, gasPrice types.AttoFIL, gasLimit types.GasUnits, method string, params ...interface{}) (cid.Cid, error)
+	MessageWait(ctx context.Context, msgCid cid.Cid, cb func(*types.Block, *types.SignedMessage, *types.MessageReceipt) error) error
+	MinerGetOwnerAddress(ctx context.Context, minerAddr address.Address) (address.Address, error)
+}
+
+// MinerSetWorkerAddress sets the worker address of the miner actor to the provided new address,
+// waits for the message to appear on chain and then sets miner.workerAddr config to the new address.
+func MinerSetWorkerAddress(
+	ctx context.Context,
+	plumbing mwapi,
+	workerAddr address.Address,
+	gasPrice types.AttoFIL,
+	gasLimit types.GasUnits,
+) error {
+
+	ret, err := plumbing.ConfigGet("mining.minerAddress")
+	if err != nil {
+		return err
+	}
+
+	minerAddr, ok := ret.(address.Address)
+	if !ok {
+		return errors.New("Problem getting miner address")
+	}
+
+	minerOwnerAddr, err := plumbing.MinerGetOwnerAddress(ctx, minerAddr)
+	if err != nil {
+		return errors.Wrap(err, "could not get miner owner address")
+	}
+
+	smsgCid, err := plumbing.MessageSend(
+		ctx,
+		minerOwnerAddr,
+		minerAddr,
+		types.ZeroAttoFIL,
+		gasPrice,
+		gasLimit,
+		"changeWorker",
+		workerAddr)
+	if err != nil {
+		return err
+	}
+
+	return plumbing.MessageWait(ctx, smsgCid, func(blk *types.Block, smsg *types.SignedMessage, receipt *types.MessageReceipt) error {
```

Comment:
Musing beyond the scope of this PR: it feels like we don't have a strong convention between returning the result of a message wait and returning a message cid that the caller can wait on.  I would like us to come up with a good one so it is obvious which approach is right given a new porcelain call like this.

Reply:
I can't really think of a satisfying answer.  I think I'd prefer always returning a CID and forcing the user to have to MessageWait every time. This is unsatisfying but it gives consistency and also the most freedom to not have to wait around for the command to finish.